src/Centipede/Filter/UrlFilter.php

```diff
@@ -4,25 +4,43 @@
 
 class UrlFilter implements FilterInterface
 {
-    private $baseUrl;
-
-    public function __construct($baseUrl)
+    public function filter($url, $previousUrl = null)
     {
-        $this->baseUrl = rtrim($baseUrl, '/');
-    }
+        // These are all the settings we'll need to recreate the url.
+        // They're also the different keys that parse_url can return.
+        $parseUrlSettings = array('scheme' => '', 'host' => '', 'path' => '', 'port' => '', 'query' => '');
 
-    public function filter($value)
-    {
-        $value = rtrim($value, '/');
+        $parseValueUrl = array_merge($parseUrlSettings, parse_url($url));
 
-        if (false !== $position = strpos($value, '#')) {
-            $value = substr($value, 0, $position);
+        if (strpos($url, '#') === 0) {
```

Comment:
I am thinking about urls with `#`.
Let's imagine we have `http://domain.com` and we are on the page `/index.html` with three links - `http://domain.com/index.html#aboutus`, `http://domain.com/contacts.html#map` and `http://domain.com/contacts.html#something_else`. 
What will be in the current impl? Will Centipede visit all three links independently? In this case, only one link should be added to the collection - `http://domain.com/contacts.html`, right?